extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -2198,4 +2214,57 @@ Runnable updateCurrentAndLatestOffsets()
       }
     };
   }
+
+  private Map<String, Object> getCurrentTotalStats() throws InterruptedException, ExecutionException, TimeoutException
+  {
+    Map<String, Object> allStats = Maps.newHashMap();
+    final List<ListenableFuture<Void>> futures = new ArrayList<>();
+
+    for (int groupId : taskGroups.keySet()) {
+      Map<String, Object> groupStats = (Map<String, Object>) allStats.get(String.valueOf(groupId));
+      if (groupStats == null) {
+        groupStats = Maps.newHashMap();
+        allStats.put(String.valueOf(groupId), groupStats);
+      }
+
+      final Map<String, Object> eGroupStats = groupStats;
+      TaskGroup group = taskGroups.get(groupId);
+      for (String taskId : group.taskIds()) {
+        futures.add(
+            Futures.transform(
+                taskClient.getMovingAveragesAsync(taskId),
+                (Function<Map<String, Object>, Void>) (currentStats) -> {
+                  eGroupStats.put(taskId, currentStats);
+                  return null;
+                }
+            )
+        );
+      }
+    }
+
+    for (int groupId : pendingCompletionTaskGroups.keySet()) {
+      Map<String, Object> groupStats = (Map<String, Object>) allStats.get(String.valueOf(groupId));
+      if (groupStats == null) {
+        groupStats = Maps.newHashMap();
+        allStats.put(String.valueOf(groupId), groupStats);
+      }
+
+      final Map<String, Object> eGroupStats = groupStats;
+      TaskGroup group = taskGroups.get(groupId);
+      for (String taskId : group.taskIds()) {
+        futures.add(
+            Futures.transform(
+                taskClient.getMovingAveragesAsync(taskId),
```

Comment:
is this really so compute intensive that parallel even helps?

Reply:
or is this based on network IO?

Either way please add a javadoc on the method

Reply:
this is network IO, the supervisor is retrieving stats from each of the tasks it manages

added javadoc